### PR TITLE
Feat: mer locking

### PR DESCRIPTION
**Pull request by codewithgun**
- MER locking program

### CODE REVIEW

File: programs/locking/src/lib.rs [hunk 1]
```diff
@@ -196,27 +158,19 @@ mod locking {
...
-            unmint_amount,
+            unlock_amount,
         )?;
 
-        emit!(EventUnStake {
-            unmint_amount,
-            token_amount: withdraw_amount,
+        emit!(EventUnlock {
+            unmint_amount: unlock_amount,
+            token_amount: unlock_amount,
```
Comment (andrewsource147) [inline programs/locking/src/lib.rs:166]: they both the same, so we just need 1 field

File: programs/locking/src/lib.rs [hunk 2]
```diff
@@ -100,70 +99,33 @@ mod locking {
...
             ),
-            mint_amount,
+            amount,
         )?;
 
-        emit!(EventStake {
-            mint_amount,
+        emit!(EventLock {
+            mint_amount: amount,
             token_amount: amount,
```
Comment (andrewsource147) [inline programs/locking/src/lib.rs:107]: They are the same, so we just need 1 field

File: programs/locking/src/lib.rs [hunk 3]
```diff
@@ -196,27 +158,19 @@ mod locking {
...
-    pub fn change_funder(ctx: Context<FunderChange>) -> Result<()> {
-        let vault = &mut ctx.accounts.vault;
-        let funder = *ctx.accounts.funder.key;
-        vault.funder = funder;
-        Ok(())
-    }
 }
 
-/// Accounts for [InitializeVault](/staking/instruction/struct.InitializeVault.html) instruction
+/// Accounts for [InitializeVault](/locking/instruction/struct.InitializeVault.html) instruction
```
Comment (andrewsource147) [inline programs/locking/src/lib.rs:173]: Pls update vault space, it is pretty small now

File: programs/locking/src/lib.rs (hunk 1 above)
Comment (codewithgun) [reply programs/locking/src/lib.rs]: Fixed

File: programs/locking/src/lib.rs (hunk 2 above)
Comment (codewithgun) [reply programs/locking/src/lib.rs]: Fixed

File: programs/locking/src/lib.rs (hunk 3 above)
Comment (codewithgun) [reply programs/locking/src/lib.rs:170]: Fixed

File: tests/staking/utils.ts [hunk 4]
```diff
@@ -7,7 +7,11 @@ import {
   TOKEN_PROGRAM_ID,
 } from "@solana/spl-token";
 
+<<<<<<< HEAD
 const program = anchor.workspace.Staking;
+=======
+const program = anchor.workspace.SingleFarming;
+>>>>>>> rename project structure
```
Comment (andrewsource147) [inline tests/staking/utils.ts:14]: Conflict here

Comment (codewithgun) [reply tests/staking/utils.ts]: Fixed

**Review by andrewsource147 — APPROVED** [review]

**Pull request merged**